lib/solarwinds_apm/support.rb

```diff
@@ -21,23 +21,21 @@
 
 if SolarWindsAPM::Config[:tag_sql]
   if defined?(Rails)
-    if Rails.version < '7'
-      require_relative 'support/swomarginalia/railtie'
-    elsif Rails.version >= '7'
-      # User has to define in their config/environments:
-      # config.active_record.query_log_tags = [
-      #   {
-      #     tracecontext: -> {
-      #       SolarWindsAPM::SWOMarginalia::Comment.traceparent
-      #     }
-      #   }
-      # ]
+
+    if Rails.version >= '7'
       SolarWindsAPM.logger.info do
-        'In Rails 7, tag tracecontext on a query by including SolarWindsAPM::SWOMarginalia::Comment.traceparent as function in config.active_record.query_log_tags.'
+        'In Rails 7, tag tracecontext on a query by including SolarWindsAPM::SWOMarginalia::Comment.traceparent as a function in config.active_record.query_log_tags. ' \
+          'In Rails >= 7.1, change the default formatter to sqlcommenter via config.active_record.query_log_tags_format = :sqlcommenter for the correct format. ' \
+          'For more info, visit https://api.rubyonrails.org/classes/ActiveRecord/QueryLogs.html'
       end
-      SolarWindsAPM.logger.info { 'For more information, please check https://api.rubyonrails.org/classes/ActiveRecord/QueryLogs.html' }
       require_relative 'support/swomarginalia/comment'
     end
+
+    if Rails.version < '7'
+      require_relative 'support/swomarginalia/railtie'
+    elsif Rails.version >= '7' && Rails.version < '7.1'
+      require_relative 'support/swomarginalia/formatter'
+    end
```

Comment:
I find the multiple checks of Rails.version >= '7' in lines 25 and 36 confusing, and the logging of usage info in lines 26-30 really obscures what the code is trying to do.

Now that you've made the great move of putting usage info in the CONFIGURATION doc, how about removing that log message, and restructure the if/else to be:

```
if rails < 7
  require railtie to achieve sql commenting
else # rails >= 7
  require comment to make the trace context available 
  if rails < 7.1
    require formatter to enforce the "sqlcommenter" style formatter
```

Reply:
Updated